### PR TITLE
lib: Filter out include paths when looking for clang paths.

### LINKED ISSUE #848: When only generating functions, type parameter is lost
We still have more issues like #835...

### Input C/C++ Header

* [ ] TODO

### Bindgen Invocation

* [ ] TODO

### Actual Results

https://treeherder.mozilla.org/logviewer.html#?job_id=117196247&repo=try&lineNumber=6725

```
error[E0243]: wrong number of type arguments: expected 1, found 0
   --> /home/worker/workspace/build/src/obj-firefox/toolkit/library/x86_64-unknown-linux-gnu/release/build/style-077f5b469d68e8b7/out/gecko/bindings.rs:526:14
    |
526 |      -> *mut nsTArray;
    |              ^^^^^^^^ expected 1 type argument
error[E0243]: wrong number of type arguments: expected 1, found 0
   --> /home/worker/workspace/build/src/obj-firefox/toolkit/library/x86_64-unknown-linux-gnu/release/build/style-077f5b469d68e8b7/out/gecko/bindings.rs:529:65
    |
529 |     pub fn Gecko_DestroyAnonymousContentList(anon_content: *mut nsTArray);
    |                                                                 ^^^^^^^^ expected 1 type argument
error[E0243]: wrong number of type arguments: expected 1, found 0
    --> /home/worker/workspace/build/src/obj-firefox/toolkit/library/x86_64-unknown-linux-gnu/release/build/style-077f5b469d68e8b7/out/gecko/bindings.rs:1073:53
     |
1073 |     pub fn Gecko_ResizeTArrayForStrings(array: *mut nsTArray, length: u32);
     |                                                     ^^^^^^^^ expected 1 type argument
error: aborting due to 3 previous errors
error: Could not compile `style`.
```

### Expected Results

`nsTArray` should have a type argument.

### PR DESCRIPTION
**Pull request by emilio**
Fixes #848

### CODE REVIEW

**Comment by emilio** [comment]
And again, all credit due to @upsuper for that amazing test-case :)

File: tests/headers/issue-848/an-include.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+template<typename T>
+class nsTArray {
+  void* mBuffer;
+};
+
+/**
+ * <div rustbindgen replaces="nsTArray"></div>
```
Comment (fitzgen) [inline tests/headers/issue-848/an-include.h]: *Super* happy you were able to create a regression test.

For posterity, I'd appreciate a comment here about how comments aren't parsed by clang for system headers, and how that breaks our annotations. Basically a summary of all the discoveries that ended up in #848's discussion.

**Pull request merged**